tests/run/named-tuples.scala

```diff
@@ -0,0 +1,107 @@
+import language.experimental.namedTuples
+import NamedTuple.*
+
+type Person = (name: String, age: Int)
+val bob = (name = "Bob", age = 33): (name: String, age: Int)
+val person2: (name: String, age: Int) = bob
+
+type Uni = (uni: Double)
+val uni = (uni = 1.0)
+val _: Uni = uni
+
+type AddressInfo = (city: String, zipCode: Int)
+val addr = (city = "Lausanne", zipCode = 1003)
+val _: AddressInfo = addr
+
+type CombinedInfo = NamedTuple.Concat[Person, AddressInfo]
+val bobWithAddr = bob ++ addr
+val _: CombinedInfo = bobWithAddr
+val _: CombinedInfo = bob ++ addr
+
+@main def Test =
+  println(bob)
+  println(bob.age)
+  println(person2.name)
+  println(bobWithAddr)
+  bob match
+    case p @ (name = "Bob", age = _) => println(p.age)
+  bob match
+    case p @ (name = "Bob", age = age) => assert(age == 33)
+  bob match
+    case p @ (name = "Peter", age = _) => println(p.age)
+    case p @ (name = "Bob", age = 0) => println(p.age)
+    case _ => println("no match")
```

Comment:
We should add some tests for creating and patter-matching nested named tuples as well.